### PR TITLE
[15.0][FIX] account_operating_unit: do not show inter-OU balance journals as invoice lines

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Please do merge without bumping the version number

Fixes: https://github.com/OCA/operating-unit/issues/617

I am not sure about adding a new boolean for this purpose:

- Normally companies won't allow setting posting entries back to draft
- The use case is not common, so adding a new column in a big table may not be good

We can delete the journal items by filtering the ones with name "OU-Balancing" or ref "Inter OU Balancing", but that criteria is not exact as long that reference could be added manually by the user, and the system will be deleting those journal items incorrectly.

@ForgeFlow

### CODE REVIEW

File: account_operating_unit/models/account_move.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class AccountMoveLine(models.Model):
     operating_unit_id = fields.Many2one(
         comodel_name="operating.unit",
     )
+    is_ou_balance = fields.Boolean()
```
Comment (JordiMForgeFlow) [inline account_operating_unit/models/account_move.py]:
```suggestion
    is_ou_balance = fields.Boolean(readonly=True)
```

Just to be safe :) The rest LGTM, I do not see other way to identify them without the flag.

Would be nice to have a migration script to fill old values

Comment (AaronHForgeFlow) [reply account_operating_unit/models/account_move.py]: Nice suggestions :) added both of them

**Review by JordiMForgeFlow — APPROVED** [review]
code review 👍🏼

File: account_operating_unit/migrations/15.0.1.3.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+def migrate(env, version):
+    if not openupgrade.column_exists(env.cr, "account_move_line", "is_ou_balance"):
+        openupgrade.logged_query(
+            env.cr,
+            """
+            ALTER TABLE account_move_line
+            ADD COLUMN IF NOT EXISTS is_ou_balance BOOLEAN DEFAULT false
+            """,
+        )
+        openupgrade.logged_query(
```
Comment (MiquelRForgeFlow) [inline account_operating_unit/migrations/15.0.1.3.0/pre-migration.py:16]:
```suggestion
        openupgrade.logged_query(
                env.cr,
                """ALTER TABLE account_move_line ALTER COLUMN is_ou_balance DROP DEFAULT""",
        )
        openupgrade.logged_query(
```

Comment (AaronHForgeFlow) [reply account_operating_unit/migrations/15.0.1.3.0/pre-migration.py:16]: Done. Thanks

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
If no more comments are added here I will be merging this soon, better to fix the bug as soon as possible.

**Comment by AaronHForgeFlow** [comment]
I think 3 weeks is enough time. We can merge this now

/ocabot merge nobump

**Comment by AaronHForgeFlow** [comment]
I will propose the forward port very soon. It is not easy, because there was a refactor in journal items boolean fields between v15 & v16

**Pull request merged**